colossus/src/main/scala/colossus/controller/OutputController.scala

```diff
@@ -285,4 +291,13 @@ trait OutputController[Input, Output] extends MasterController[Input, Output] {
     }
   }
 
+  def idleCheck(period: Duration): Unit = {
```

Comment:
Possibly a totally inane comment, but figured I'd let it rip.

I just want to make sure we didn't introduce a race condition. Before, the only function that was mutating the 'waitingToSend' queue was 'checkQueue'.  The scenario I was dreaming up was where both checkQueue and idleCheck were invoked around the same time, and this function says "yep..the head is timed out", and before it pops, checkQueue rushes in and pops the queue.  Then this pops the queue.  Oops.  We would have just popped a non timed out message.

However, after tracking down the invocations of checkeQueue(and its callers callers), it seems like they are all bound by the Worker, meaning that they are called only when the Worker initiates the call chain after it receives a message.  Since that is threadsafe, we are safe here as well.  

I think that covers it, but @DanSimon there are no other places where these functions get invoked, right?